### PR TITLE
marsRover15: Displays the user's name on the home page if logged in

### PR DESCRIPTION
**Pull request by HeatherJLewis**
**Description**
Display the user's name on the home page if logged in

**Changes**
* adds `/getUsername` which serves username as JSON
* renders username on home page if logged in

### CODE REVIEW

File: config/passport.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const JwtStrategy = require('passport-jwt').Strategy;
 const { PRIVATE_KEY } = require('../config/apiCredentials');
 
 const cookieExtractor = (request) => {
-	var token = null;
+	let token = null;
```
Comment (Kenneth-Siu) [inline config/passport.js]: Ah yes, this has highlighted something I accidentally glossed over.

`let` alert! :alarm: :alarm:

I think we can get away without needing a let here and just returning the token's value at each point where we currently set it.

File: config/passport.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ const JwtStrategy = require('passport-jwt').Strategy;
 const { PRIVATE_KEY } = require('../config/apiCredentials');
 
 const cookieExtractor = (request) => {
```
Comment (Kenneth-Siu) [inline config/passport.js]: I wonder if this function could be renamed. Cookie extractor sounds like it extracts the cookie, whereas we're really extracting the jwt.

File: server/middleware/getUsername.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const jwtDecode = require('jwt-decode');
+
+const getUsername = (request, response) => {
+	const accessToken = request.headers.cookie.split('=')[1];
```
Comment (Kenneth-Siu) [inline server/middleware/getUsername.js]: I guess you wrote this before the magic of cookie-parser

**Review by Kenneth-Siu — COMMENTED** [review]
Overall looks good! A few tiny code tidying up bits.

I wonder what NASA would think if we ever published this and we wrote on our page "welcome to NASA" 🤣

File: server/middleware/getUsername.js (hunk 3 above)
Comment (HeatherJLewis) [reply server/middleware/getUsername.js]: Hmm. That should have been changed to cookie-parser. Will go and check.

Comment (HeatherJLewis) [reply server/middleware/getUsername.js]: Sorted! :)

File: config/passport.js (hunk 2 above)
Comment (HeatherJLewis) [reply config/passport.js]: Done

File: config/passport.js (hunk 1 above)
Comment (HeatherJLewis) [reply config/passport.js]: I think I've sorted this out.

**Review by Kenneth-Siu — APPROVED** [review]
ace 💯

**Pull request merged**